tools/chplcheck/src/rules.py

```diff
@@ -331,8 +334,14 @@ def MisleadingIndentation(context, root):
             yield from MisleadingIndentation(context, child)
 
             if prev is not None:
-                if child.location().start()[1] == prev.location().start()[1]:
-                    yield AdvancedRuleResult(child, prevloop)
+                loc = child.location()
+                prev_loc = prev.location()
+                if loc.start()[1] == prev_loc.start()[1]:
+                    dedent = prev_loc.start()[1] - prevloop.location().start()[1]
+                    start = (loc.start()[0], loc.start()[1] - dedent)
+                    text = range_to_text(loc, lines)
+                    fixit = Fixit.build(Edit(loc.path(), start, loc.end(), text))
+                    yield AdvancedRuleResult(child, prevloop, fixits=[fixit])
```

Comment:
I think you need to guard against inane code like this:

```Chapel
   for i in 1..10 do
writeln("Hello");
writeln("World");
```

This code should trigger misleading indentation but likely won't get fixed the way you expect. Consider just replacing the whole line with "new depth + line after old indent". 

```Python
if loc.start()[1] == prev_loc.start()[1]:
    line_start = (loc.start()[0], 1)
    parent_indent = prevloop.location().start()[1]
    text = " " * parent_indent + range_to_text(loc, lines)
    fixit = Fixit.build(Edit(loc.path(), line_start, loc.end(), text))
```

This also doesn't work for multiline code; I suggest just skipping fixits in that case, since the user's indentation policy for multi-line stmts may vary.

Reply:
I have implemented this (as well as adding tests for the cases you mentioned)